src/pages/home/index.js

```diff
@@ -0,0 +1,582 @@
+  }, [selectedCriteria]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  React.useEffect(() => {
+    getDistricts();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedState]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  React.useEffect(() => {
+    if (selectedDistrict || selectedPin) {
+      getVaccines();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedDistrict, selectedPin, date, selectedCriteria]);
+
+  const generateURL = (
+    _sub = null,
+    _pin = null,
+    _district = null,
+    _calendar = false
+  ) => {
+    if (_sub) {
+      return `${BASE_URL}${_sub}`;
+    }
+
+    let sub;
+    const formattedDate = typeof date === "string" ? date : formatDate(date);
+
+    if (_pin && _calendar) {
+      sub = `/appointment/sessions/public/calendarByPin?pincode=${_pin}&date=${formattedDate}`;
+    } else if (_pin) {
+      sub = `/appointment/sessions/public/findByPin?pincode=${_pin}&date=${formattedDate}`;
+    } else if (_district && _calendar) {
+      sub = `/appointment/sessions/public/calendarByDistrict?district_id=${_district.district_id}&date=${formattedDate}`;
+    } else if (_district) {
+      sub = `/appointment/sessions/public/findByDistrict?district_id=${_district.district_id}&date=${formattedDate}`;
+    }
+
+    const url = `${BASE_URL}${sub}`;
+    return url;
+  };
+
+  const getStates = async () => {
+    try {
+      setLoadingStates(true);
+      setError(false);
+      setVaccines(null);
+      setSelectedState(null);
+      setSelectedDistrict(null);
+      const response = await fetch(generateURL("/admin/location/states"));
+      const data = await response.json();
+      setLoadingStates(false);
+      setStates(data.states);
+    } catch (err) {
+      setLoadingStates(false);
+      setError(true);
+    }
+  };
+
+  const getDistricts = async () => {
+    try {
+      setVaccines(null);
+      setSelectedDistrict(null);
+      setError(false);
+      if (selectedState && selectedState.state_id) {
+        setSelectedDistrict(null);
+        setLoadingDistricts(true);
+        const response = await fetch(
+          generateURL(`/admin/location/districts/${selectedState.state_id}`)
+        );
+        const data = await response.json();
+        setLoadingDistricts(false);
+        setDistricts(data.districts);
+      }
+    } catch (error) {
+      setLoadingDistricts(false);
+      setError(true);
+    }
+  };
+
+  const formatDate = (_date, format = "in") => {
+    const month = _date.getMonth() + 1;
+    const day = _date.getDate();
+    const year = _date.getFullYear();
+    if (format === "us") {
+      const formatted = `${year}-${month < 10 ? "0" + month : month}-${
+        day < 10 ? "0" + day : day
+      }`;
+      return formatted;
+    }
+    return `${day < 10 ? "0" + day : day}-${
+      month < 10 ? "0" + month : month
+    }-${year}`;
+  };
+
+  const refactor = (_vaccine) => {
```

Comment:
What is this refactor function doing? Please explain

Reply:
@stephin007, there are two types of responses that the application receives from the vaccines endpoint: "Find by district/pincode" and "Find by district/pincdoe & date". Before the app renders either of these responses, they need to be converted into an object that follows only one, fixed structure ( as opposed to the two, legacy structures ). The `refactor` method performs this "conversion"